src/gc/unix/cgroup.cpp

```diff
@@ -112,16 +121,149 @@ class CGroup
     }
 
     static bool GetCpuLimit(uint32_t *val)
+    {
+        if (GetCGroup1CpuLimit(val))
+        {
+            return true;
+        }
+
+        if (GetCGroup2CpuLimit(val))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+private:
+    static bool IsCGroup1MemorySubsystem(const char *strTok){
+        return strcmp("memory", strTok) == 0;
+    }
+
+    static bool IsCGroup1CpuSubsystem(const char *strTok){
+        return strcmp("cpu", strTok) == 0;
+    }
+
+    static bool GetCGroup2CpuLimit(uint32_t *val)
+    {
+        char *line = nullptr;
+        char *filename = nullptr;
+        FILE *file = nullptr;
+        char *max_quota_string = nullptr;
+        char *period_string = nullptr;
+
+        size_t lineLen = 0;
+        int sscanRet = 0;
+        long long max_quota = 0;
+        long long period = 0;
+        double cpu_count = 0;
+
+        bool result = false;
+
+        if (s_cpu_cgroup_path == nullptr)
+        {
+            return false;
+        }
+
+        size_t filename_len = strlen(s_cpu_cgroup_path) + strlen(CGROUP2_CPU_MAX_FILENAME);
+        filename = (char*)malloc(filename_len + 1);
+        if (filename == nullptr)
+        {
+            goto done;
+        }
+
+        strcpy(filename, s_cpu_cgroup_path);
+        strcat(filename, CGROUP2_CPU_MAX_FILENAME);
+
+        file = fopen(filename, "r");
+        if (file == nullptr)
+        {
+            goto done;
+        }
+
+        if (getline(&line, &lineLen, file) == -1)
+        {
+            goto done;
+        }
+
+        // The expected format is:
+        //     $MAX $PERIOD
+        // Where "$MAX" may be the string literal "max"
+
+        max_quota_string = (char*) malloc(lineLen + 1);
+        if (max_quota_string == nullptr)
+        {
+            goto done;
+        }
+        period_string = (char*) malloc(lineLen + 1);
+        if (period_string == nullptr)
+        {
+            goto done;
+        }
+
+        sscanRet = sscanf(line, "%s %s", max_quota_string, period_string);
+        if (sscanRet != 2)
+        {
+            assert(!"Unable to parse " CGROUP2_CPU_MAX_FILENAME " file contents with sscanf.");
+            goto done;
+        }
```

Comment:
This is dangerous: you're writing an unbounded amount of characters to both `max_quota_string` and `period_string`.  Consider just slicing `line` instead by looking for `' '` with `memchr()` and writing a '\0' there.  Saves two `malloc()` calls too:

```C
char *space = memchr(line, ' ', lineLen);
if (!space) goto done;

max_quota_string = line;
*line = '\0';
period_string = space + 1;
```